lcls_naming_tool/lcls_naming_tool.py

```diff
@@ -0,0 +1,309 @@
+#!/usr/bin/env python3
+
+'''
+    LCLS Naming Tool
+
+    A tool for checking the form and content (but not meaning) of PV names with
+    respect to the LCLS naming convention.
+
+    Functional component source letters and beampath numbers come from 
+    https://docs.google.com/drawings/d/1Sg7pT2TlKq8PgM7UstmJJBLKeqROQeDWbkD0sDoTDdY/edit
+
+    A valid PV name is in the format FFFFF:GGG:CCC:NN:XXXX, where GGG and NN are
+    optional and can be omitted. The XXXX is omitted for device names.
+
+    All permutations of valid PV and device names according to 
+    LCLS_Photon_Source_and_Systems_Nomenclature.pdf. Names ending in XXXX are 
+    PV names. Without XXXX are device names.
+
+        FFFFF:CCC
+        FFFFF:CCC:NN
+        FFFFF:CCC:XXXX
+        FFFFF:CCC:NN:XXXX
+        FFFFF:GGG:CCC
+        FFFFF:GGG:CCC:NN
+        FFFFF:GGG:CCC:XXXX
+        FFFFF:GGG:CCC:NN:XXXX
+
+    This script only checks the validity of PV and device names.
+'''
+
+import argparse
+import subprocess
+import sys
+import json
+import re
+
+
+fc_dict = {}
+fg_dict = {}
+ccc_dict = {}
+beam_sources = {'K': '', 'L': ''}
+beam_numbers = {'0': '', '1': '', '2': '', '3': '', '4': '', '5': ''}
+
+
+def get_version():
```

Comment:
In a later effort, we should refactor this to use our standard setuptools_scm version processing.   This will currently just output the most recent tag, and doesn't actually check if the version contained matches that tag.

Reply:
Saved this comment link for the next PR.